kardinal-cli/cmd/root.go

```diff
@@ -227,16 +224,46 @@ func parseKubernetesManifestFile(kubernetesManifestFile string) ([]api_types.Ser
 		switch obj := obj.(type) {
 		case *corev1.Service:
 			service := obj
-			serviceConfigs[index/2].Service = *service
+			serviceName := getObjectName(service.GetObjectMeta().(*metav1.ObjectMeta))
+			_, ok := serviceConfigs[serviceName]
+			if !ok {
+				serviceConfigs[serviceName] = &api_types.ServiceConfig{
+					Service: *service,
+				}
+			} else {
+				serviceConfigs[serviceName].Service = *service
+			}
 		case *appv1.Deployment:
 			deployment := obj
-			serviceConfigs[index/2].Deployment = *deployment
+			deploymentName := getObjectName(deployment.GetObjectMeta().(*metav1.ObjectMeta))
+			_, ok := serviceConfigs[deploymentName]
+			if !ok {
+				serviceConfigs[deploymentName] = &api_types.ServiceConfig{
+					Deployment: *deployment,
+				}
+			} else {
+				serviceConfigs[deploymentName].Deployment = *deployment
+			}
 		default:
 			return nil, stacktrace.NewError("An error occurred parsing the manifest because of an unsupported kubernetes type")
 		}
 	}
 
-	return serviceConfigs, nil
+	finalServiceConfigs := []api_types.ServiceConfig{}
+	for _, serviceConfig := range serviceConfigs {
+		finalServiceConfigs = append(finalServiceConfigs, *serviceConfig)
+	}
+
+	return finalServiceConfigs, nil
+}
+
+func getObjectName(obj *metav1.ObjectMeta) string {
```

Comment:
Can we add a comment to make this behavior explicit? 

it got me for a little bit - my understanding is that when we parse, we are using the label `app` to pair `ServiceSpec` and `DeploymentSpecs` into the same `ServiceConfig`. 

This means that on the manifest this label is required on the `DeploymentSpec` and needs to be the same as the `ServiceSpec` name - which I didn't initially have.